### ISSUE TITLE
MoSDeF Input Testing Matrix

### ISSUE THREAD

**Issue by ahy3nz** [ahy3nz/coMMParE#13]
* We have a lot of structure/chemical formulas as reference systems
* We will soon have access to multiple general force fields (OPLS and GAFF) for atomtyping and parametrization

We need a somewhat-elegant and flexible way to test all these possible combinations. My first thought is just using `it.product` within `mm_tests/mosdef_inputs` to build + atomtype/parametrize all the different permutations and then shoot them all off to different engines

**Comment by mattwthompson** [comment]
What are the permutations, structures * force fields * engines?

**Comment by ahy3nz** [comment]
Yeah I was thinking something like that. I'm not sure what the actual usefulness is or if it'd just be something neat to try. In reality, I think [basic unit tests](https://github.com/ctk3b/validate/tree/master/validate/tests/gromacs/unit_tests) that solely look at molecular models of a single function are probably the best at debugging force field translation

**Comment by mattwthompson** [comment]
I think each layer has a huge amount of utility

* Running over a bunch of structures gives us an idea of what chemistries give us problems (at each step in the process)

* Running over force fields gives us an idea of how much is covered

* Running over engines would let us diagnose issues with conversions or, potentially, errors in the engines themselves, particularly with exotic potentials/etc.

The last layer would be comparing the output energies (or other physical properties) to reference data. For example, we could probably write out a monumentally intricate molecule, type it in GAFFF, and run it in a bunch of engines, but are the results physical/reasonable/viablet/trustable? Probably outside the scope of this project but it's the sort of project I would like somebody to pick up. Basically I'm suggesting an updated version of [this](https://pubs.acs.org/doi/abs/10.1021/ct200731v) project (http://virtualchemistry.org/).
